src/Network/Wai/Log/Options.hs

```diff
@@ -52,14 +55,16 @@ data ResponseTime = ResponseTime {
 -- { logLevel = 'LogInfo'
 -- , logRequest = 'defaultLogRequest'
 -- , logResponse = 'defaultLogResponse'
+-- , logGetRequestId = 'const nextRandom'
 -- }
 -- @
-defaultOptions :: Options
+defaultOptions :: Options UUID
```

Comment:
One thing I'm bit worried about it that you won't be able to just change `logGetRequestID` using record update, if the function doesn't use `UUID`. Can you try that? I believe something like this would fail:

```
defaultOptions { logGetRequestId = const 5 :: (Int) }
```

In such case, I would consider another function that actually takes , something like `mkDefaultOptions :: (Request -> IO id) -> Options id` so we can have actually construct default options of arbitrary `id` (maybe think of more suitable name though).

Reply:
The reason this doesn't work is because other functions in the record are bound by the same type variable `id`. Specifically `logRequest` and `logResponse` so for this to work you need to update them as well in the record update.
```hs
:type defaultOptions { logGetRequestId = const (pure 5) :: Request -> IO Int, logRequest = undefined, logResponse = undefined }
Options Int
```
This works.